modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -135,6 +150,45 @@ impl ClientState {
         Some(2 * self.trusting_period / 3)
     }
 
+    pub fn chain_id(&self) -> ChainId {
+        self.chain_id.clone()
+    }
+
+    pub fn client_type(&self) -> ClientType {
+        ClientType::Tendermint
+    }
+
+    pub fn client_def(&self) -> TendermintClient<G>
+    where
+        G: GlobalDefs,
+    {
+        TendermintClient::default()
+    }
+
+    pub fn frozen_height(&self) -> Option<Height> {
+        self.frozen_height
+    }
+
+    pub fn upgrade(
+        mut self,
+        upgrade_height: Height,
+        upgrade_options: UpgradeOptions,
+        chain_id: ChainId,
+    ) -> Self {
+        // Reset custom fields to zero values
+        self.trusting_period = ZERO_DURATION;
+        self.trust_level = TrustThreshold::ZERO;
+        self.frozen_height = None;
+        self.max_clock_drift = ZERO_DURATION;
+
+        // Upgrade the client state
+        self.latest_height = upgrade_height;
+        self.unbonding_period = upgrade_options.unbonding_period;
+        self.chain_id = chain_id;
```

Comment:
does chain id change every time?

Reply:
I don't know. @Wizdave97 ?

Reply:
Yeah it could, if the chain updates it's consensus mechanism or has a fork maybe.